library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerView.java

```diff
@@ -1055,26 +1070,20 @@ private boolean setArtworkFromMetadata(Metadata metadata) {
       if (metadataEntry instanceof ApicFrame) {
         byte[] bitmapData = ((ApicFrame) metadataEntry).pictureData;
         Bitmap bitmap = BitmapFactory.decodeByteArray(bitmapData, 0, bitmapData.length);
-        return setArtworkFromBitmap(bitmap);
+        return setDrawableArtwork(new BitmapDrawable(getResources(), bitmap));
       }
     }
     return false;
   }
 
-  private boolean setArtworkFromBitmap(Bitmap bitmap) {
-    if (bitmap != null) {
-      int bitmapWidth = bitmap.getWidth();
-      int bitmapHeight = bitmap.getHeight();
-      if (bitmapWidth > 0 && bitmapHeight > 0) {
-        if (contentFrame != null) {
-          contentFrame.setAspectRatio((float) bitmapWidth / bitmapHeight);
-        }
-        artworkView.setImageBitmap(bitmap);
-        artworkView.setVisibility(VISIBLE);
-        return true;
+  private boolean setDrawableArtwork(Drawable drawable) {
+    if(drawable != null) {
+      artworkView.setImageDrawable(drawable);
+      if(contentFrame != null) {
+        contentFrame.setAspectRatio(0);
```

Comment:
Removing the logic for setting the aspect ratio seems to break displaying bitmap artwork (artwork is now stretched). It appears it's possible to fix this either by:

1. Bringing back setting the aspect ratio correctly, but using `getIntrinsicWidth` and `getIntrinsicHeight` on the `Drawable`.
1. In `exo_simple_player_view.xml`, changing `exo_artwork`'s `scaleType` to `fitCenter`.

I think my preference would be the first one, since it's most similar to what we had before, and use of the `contentFrame` for controlling aspect ratio is more consistent with what happens for video.

I'm unsure whether it breaks your use case for non-bitmap drawables though. Could you take a look at this?

No need to update the pull request, but let me know whether use of `getIntrinsicWidth` and `getIntrinsicHeight` works for you. Thanks!

Reply:
We'll go with using the intrinsic dimensions for now, but feel free to send another pull request changing that if it doesn't work for some use cases.

Reply:
Hi, sorry for the late reply. It won't break my case, and I think it would be fine, but my idea was to indeed leave this to the developer to set the `scaleType` in the XML. Anyway, the 1st solution works for my case. Thanks!